compiler/type-hint.cpp

```diff
@@ -337,6 +337,19 @@ const TypeHint *TypeHintShape::create(std::vector<std::pair<std::string, const T
   );
 }
 
+std::map<std::int64_t, std::string> TypeHintShape::shape_keys_storage_;
+std::mutex TypeHintShape::shape_keys_storage_lock_;
+
+void TypeHintShape::set_shape_key(std::int64_t key_hash, std::string_view key) noexcept {
+  std::lock_guard lock{shape_keys_storage_lock_};
+  shape_keys_storage_.emplace(key_hash, key);
+}
+
+const std::map<std::int64_t, std::string> &TypeHintShape::get_shape_keys_storage() noexcept {
+  std::lock_guard lock{shape_keys_storage_lock_};
```

Comment:
We don't need any lock here:
1) accessing the filled storage is done much later that filling it
2) even if (1) is not true, this lock wouldn't actually lock reading access to it from the calling side

Reply:
Done